client/scripts/controllers/chain/cosmos/chain.ts

```diff
@@ -72,36 +92,84 @@ class CosmosChain implements IChainModule<CosmosToken, CosmosAccount> {
-
-    this._api = new CosmosApi(wsUrl, restUrl);
+    this._url = node.url;
+
+    console.log(`Starting REST API at ${this._url}...`);
+
+    console.log('cosmjs api');
+    // TODO: configure broadcast mode
+    this._api = LcdClient.withExtensions(
+      { apiUrl: this._url },
+      setupAuthExtension,
+      setupGovExtension,
+      setupStakingExtension,
+      setupBankExtension,
+      setupSupplyExtension,
+    );
     if (this.app.chain.networkStatus === ApiStatus.Disconnected) {
       this.app.chain.networkStatus = ApiStatus.Connecting;
     }
-    await this._api.init((header) => {
-      this._blocktimeHelper.stamp(moment(header.time));
-      this.app.chain.block.height = +header.height;
-      m.redraw();
-    });
+    const nodeInfo = await this._api.nodeInfo();
+    this._chainId = nodeInfo.node_info.network;
+    console.log(`chain id: ${this._chainId}`);
     this.app.chain.networkStatus = ApiStatus.Connected;
-    const { bonded_tokens } = await this._api.query.pool();
-    this._staked = +bonded_tokens;
-    const { bond_denom } = await this._api.query.stakingParameters();
-    this._denom = bond_denom;          // uatom
-    this._chainId = this._api.chainId; // cosmoshub-2
+
+    // Poll for new block immediately and then every 2s
+    const fetchBlockJob = async () => {
+      const block = await this._api.blocksLatest();
+      const height = +block.block.header.height;
+      if (height > this.app.chain.block.height) {
+        const time = moment(block.block.header.time);
+        this._blocktimeHelper.stamp(moment(time));
+        this.app.chain.block.height = height;
+        m.redraw();
+      }
+    };
+    await fetchBlockJob();
+    this._blockSubscription = setInterval(fetchBlockJob, 2000);
+
+    const { result: { bonded_tokens } } = await this._api.staking.pool();
+    this._staked = this.coins(new BN(bonded_tokens));
+    const { result: { bond_denom } } = await this._api.staking.parameters();
+    this._denom = bond_denom;
     m.redraw();
   }
 
   public async deinit(): Promise<void> {
     this.app.chain.networkStatus = ApiStatus.Disconnected;
-    if (this._api) this._api.deinit();
+    if (this._blockSubscription) {
+      clearInterval(this._blockSubscription);
+    }
+  }
+
+  public async sendTx(account: CosmosAccount, tx: EncodeObject): Promise<string> {
+    // TODO: error handling
+    const wallets = this.app.wallets.availableWallets(ChainBase.CosmosSDK);
+    if (!wallets) throw new Error('No cosmos wallet found');
+
+    // TODO: support multiple wallets
+    const wallet = wallets[0] as KeplrWebWalletController;
+    const client = await wallet.getClient(this.app.chain.meta.url, account.address);
+
+    // these parameters will be overridden by the client
+    // TODO: can it be simulated?
+    const DEFAULT_FEE: StdFee = {
+      gas: '180000',
+      amount: [{ amount: (2.5e-8).toFixed(9), denom: this.denom }]
```

Comment:
The amount of a coin is an integer string. It is expressed in the base denom, e.g. "uatom". The value here ('0.000000025') is invalid for this field.

Reply:
From CosmJS 0.26 on you can calculate the fee from gas limit and gas price:

```
  const gasPrice = GasPrice.fromString("0.025ucosm");
  const DEFAULT_FEE = calculateFee(180000, gasPrice);
```

Reply:
Cool -- will fix this once 0.26 lands.